contracts/Aavegotchi/facets/ItemsFacet.sol

```diff
@@ -182,38 +190,76 @@ contract ItemsFacet is Modifiers {
+    ///@dev A wearable cannot be equipped in the wrong slot
+    ///@param _tokenId The identifier of the aavegotchi to make changes to
+    ///@param _wearablesToEquip An array containing the identifiers of the wearables to equip
+    ///@param _commitmentIds An array containing the identifiers of the deposited wearables to equip
+    function equipDelegatedWearables(
+        uint256 _tokenId,
+        uint16[EQUIPPED_WEARABLE_SLOTS] calldata _wearablesToEquip,
+        uint256[EQUIPPED_WEARABLE_SLOTS] calldata _commitmentIds
+    )
+        external
+    {
+        
+        _equipWearables(_tokenId, _wearablesToEquip, _commitmentIds);
+    }
+
+
+    function _equipWearables(
+        uint256 _tokenId,
+        uint16[EQUIPPED_WEARABLE_SLOTS] calldata _wearablesToEquip,
+        uint256[EQUIPPED_WEARABLE_SLOTS] memory _commitmentIdsToEquip
+    )
+        internal
         onlyAavegotchiOwner(_tokenId)
         onlyUnlocked(_tokenId)
     {
         Aavegotchi storage aavegotchi = s.aavegotchis[_tokenId];
         require(aavegotchi.status == LibAavegotchi.STATUS_AAVEGOTCHI, "LibAavegotchi: Only valid for AG");
-        emit EquipWearables(_tokenId, aavegotchi.equippedWearables, _wearablesToEquip);
-
-        address sender = LibMeta.msgSender();
+        emit EquipWearables(_tokenId, s.aavegotchis[_tokenId].equippedWearables, _wearablesToEquip);
+        emit EquipDelegatedWearables(_tokenId, s.gotchiEquippedItemsInfo[_tokenId].equippedCommitmentIds, _commitmentIdsToEquip);
+        GotchiEquippedCommitmentsInfo storage _gotchiInfo = s.gotchiEquippedItemsInfo[_tokenId];
 
         for (uint256 slot; slot < EQUIPPED_WEARABLE_SLOTS; slot++) {
+            
             uint256 toEquipId = _wearablesToEquip[slot];
             uint256 existingEquippedWearableId = aavegotchi.equippedWearables[slot];
+            bool _sameWearablesIds = toEquipId == existingEquippedWearableId;
+
+            uint256 _commitmentIdToEquip = _commitmentIdsToEquip[slot];
+            uint256 _existingEquippedCommitmentId = _gotchiInfo.equippedCommitmentIds[slot];
 
             //If the new wearable value is equal to the current equipped wearable in that slot
             //do nothing
-            if (toEquipId == existingEquippedWearableId) {
+            if (_sameWearablesIds && _existingEquippedCommitmentId == _commitmentIdToEquip) {
                 continue;
             }
 
             //Equips new wearable (or sets to 0)
             aavegotchi.equippedWearables[slot] = uint16(toEquipId);
 
             //If a wearable was equipped in this slot and can be transferred, transfer back to owner.
-
+            
             if (existingEquippedWearableId != 0 && s.itemTypes[existingEquippedWearableId].canBeTransferred) {
+                if (_sameWearablesIds && (slot == LibItems.WEARABLE_SLOT_HAND_LEFT || slot == LibItems.WEARABLE_SLOT_HAND_RIGHT)) {
```

Comment:
What's this check for? @ernanirst

Reply:
I checked with @karacurt and I think we found an opportunity for improvement.

In short, the challenge of this part of the code is that we are calling `removeFromParent` which reverts on the following `if`:
```sol
if (
  aavegotchi.equippedWearables[LibItems.WEARABLE_SLOT_HAND_LEFT] == _id &&
  aavegotchi.equippedWearables[LibItems.WEARABLE_SLOT_HAND_RIGHT] == _id
) {
  revert("LibItems: Can't hold 1 item in both hands");
}
```
In the previous `equipWearables` implementation, we would never reach this piece of code because we always skipped when the tokenId removed and equipped were the same. However, with the new `equipDelegatedWearables` function, we can have the same tokenIds with a different `commitmentId` (`depositId`).

Therefore, to prevent `removeFromParent` from reverting, we delete `aavegotchi.equippedWearables[slot]` (effectively unequipping that wearable), and we only assign a new wearable to that slot later.

This is only one approach, an alternative could be to pass the `commitmentId` as a parameter to `removeFromParent`, and check the `commitmentId` to not revert as well. However, modifying the signature of `removeFromParent` will increase the scope of the changes (we will have to modify multiple parts of the code).